### PR TITLE
added routes for ticket number 21

### PR DESCRIPTION
**Pull request by mrogers23458**

### CODE REVIEW

File: controllers/api/forum-routes.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const router = require('express').Router();
+const { User , Post } = require('../../models');
+
+//gets all forum posts
+router.get("/", async (req, res) => {
```
Comment (Myuze) [inline controllers/api/forum-routes.js:5]:
```suggestion
router.get('/', async (req, res) => {
```

File: controllers/api/forum-routes.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    const forumData = await Post.findAll({
+      include: {
+        model: 'User',
+        attributes: [ 'id', 'username' ]
+      }
+    }).catch((err) => { 
+      res.json(err);
+    });
+      const forumPosts = forumData.map((fPost) => fPost.get({ plain: true }));
+    res.render("display", { forumPosts });
```
Comment (Myuze) [inline controllers/api/forum-routes.js:15]:
```suggestion
    res.render('display', { forumPosts });
```

File: controllers/api/forum-routes.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  });
+
+// gets one forum post by id
+router.get("/:id", async (req, res) => {
```
Comment (Myuze) [inline controllers/api/forum-routes.js:19]:
```suggestion
router.get('/:id', async (req, res) => {
```

File: controllers/api/forum-routes.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    const { user } = req.session;
+    const  forumPost = await Post.findOne({
+        where: {
+            id: req.params.id
+        }
+    }).catch((err) => { 
+      res.json(err);
+    });
+
+    res.render("display", { user, forumPost } );
```
Comment (Myuze) [inline controllers/api/forum-routes.js:29]:
```suggestion
    res.render('display', { user, forumPost } );
```

File: controllers/api/forum-routes.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  });
+
+  module.exports = router
```
Comment (Myuze) [inline controllers/api/forum-routes.js:32]:
```suggestion
  module.exports = router
  
```

File: controllers/globalController.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+const router = require('express').Router();
+const { User, Post } = require("../models")
+
+router.get("/", async (req, res) => {
```
Comment (Myuze) [inline controllers/globalController.js:4]:
```suggestion
router.get('/', async (req, res) => {
```

File: controllers/globalController.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    const user = req.session
+    const postData = await Post.findAll().catch((err) => {
+        res.json(err)
+    })
+
+    const posts = postData.map((post).get({ plain: true }));
+    console.log(posts)
+
+    res.render("home", { posts, user });
```
Comment (Myuze) [inline controllers/globalController.js:13]:
```suggestion
    res.render('home', { posts, user });
```

File: controllers/viewControllers/userViewsController.js [hunk 8]
```diff
@@ -0,0 +1,8 @@
+const router = require('express').Router();
+const { User, Post } = require('../../models');
+
+router.get("/login", (req, res) => {
+
+})
+
+module.exports = router;
```
Comment (Myuze) [inline controllers/viewControllers/userViewsController.js:8]:
```suggestion
module.exports = router;

```

**Review by Myuze — CHANGES_REQUESTED** [review]
Gonna share what I also mentioned in Julian's review; So for quote usage, I suggest sticking with single quotes for strings and escape apostrophes within JS (this does not apply to formal JSON), as most of the time it's more likely that double quotes will be used within a string. This is a preference thing, but we need to be consistent.

Also, for route file names let's keep it consistent. I'm up for either camelCase or dash-separated-lowercase, but not both. It just happend that the server code we reused for the project already was using dash-separated-lowercase, and we are already using camelCase for variable names.

**Review by mrogers23458 — COMMENTED** [review]
I'm not sure how to remove the updates to globalControllers, but somehow I'm changing that even after a fresh pull from git main.

**Pull request closed without merging**